### ISSUE TITLE
Portrait images rotated on iOS 13.4

### ISSUE THREAD

**Issue by SamLoy** [Mawi137/ngx-image-cropper#306]
Hi, 

Looking through the issue history of the project it looks like this has been fixed a while back. Although it would appear that iOS 13.4 has broken it. 

[https://stackblitz.com/edit/image-cropper](https://stackblitz.com/edit/image-cropper)

If i upload an image taken straight from the camera or upload an image from my library on iOS 13.4 i get my image rotated.

![Image from iOS](https://user-images.githubusercontent.com/9429328/78801821-0c434c80-79b5-11ea-9e9a-ff079e5e42bb.png)

I do have some code ready for correcting the image after it has been returned by image-cropper, but I think it will be confusing to my users that their image appears rotated on the cropper. 

I tested the same on Android and iOS 13.3.1 and all works fine 👍 

I also tried uploading the image taken from my iOS 13.4 phone and uploading it from 13.3.1 and it worked fine 👍 . Although that could be where Airdrop and Whatsapp strip out the exif..

Any ideas what might have happened here? 

Thanks

Sam

**Comment by SamLoy** [comment]
After some experimenting i found the easiest workaround for now is just to sort out the exif orientation beforehand before the file is passed to ngx-image-cropper. 

`<image-cropper [imageFile]="alreadyOritentated"></image-cropper>`

I used blueimp/Javascript-Load-Image for this: 

`resolveImageOrientation(file: File): Promise<File> {
        return new Promise<File>((resolve, reject) => {
            try {
                loadImage(
                    file,
                    (canvas: HTMLCanvasElement) => {
                        canvas.toBlob(blob => resolve(this.blobToFile(blob, file.name)), file.type, 1);
                    },
                    { maxWidth: 1000, maxHeight: 1000, canvas: true, orientation: true }
                )
            } catch (e) {
                console.error(e);
                reject(e);
            }
        });
    }

blobToFile(theBlob: Blob, fileName: string): File {
        const b: any = theBlob;
        //A Blob() is almost a File() - it's just missing the two properties below which we will add
        if (!b.lastModifiedDate) {
            b.lastModifiedDate = new Date();
        }
        if (!b.name) {
            b.name = fileName;
        }

        //Cast to a File() type
        return <File>theBlob;
    }`

**Comment by Mawi137** [comment]
I don’t seem to have that issue when I take a photo with my iPhone and pass it to the image cropper. The image cropper already reads out the exit information and transforms the image accordingly.

**Comment by Mawi137** [comment]
Although it seems I’m still on 13.3.1

**Comment by Mawi137** [comment]
Just updated to the latest iOS and you are correct. With portrait images the rotation is not correct.

**Comment by Mawi137** [comment]
I found this in the lib you posted. Appearanly Safari already rotates the images automatically and it seems since 13.4.1 that feature is also in Chrome for iOS.

```
var testImageURL =
      'data:image/jpeg;base64,/9j/4QAiRXhpZgAATU0AKgAAAAgAAQESAAMAAAABAAYAAAA' +
      'AAAD/2wCEAAEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBA' +
      'QEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQE' +
      'BAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAf/AABEIAAEAAgMBEQACEQEDEQH/x' +
      'ABKAAEAAAAAAAAAAAAAAAAAAAALEAEAAAAAAAAAAAAAAAAAAAAAAQEAAAAAAAAAAAAAAAA' +
      'AAAAAEQEAAAAAAAAAAAAAAAAAAAAA/9oADAMBAAIRAxEAPwA/8H//2Q=='
    var img = document.createElement('img')
    img.onload = function () {
      // Check if browser supports automatic image orientation:
      loadImage.orientation = img.width === 1 && img.height === 2
    }
    img.src = testImageURL
```
What they do is take an image of 2 by 1 all black with exif rotation 6. If the image is already flipped onload, then the browser support automatic rotation of the image. So I'll have to implement this in the image-cropper as well.

**Comment by Mawi137** [comment]
Just implemented the fix. Can you verify that it works here: https://image-cropper.stackblitz.io 
If so, I'll create a new release tomorrow.

**Comment by SamLoy** [comment]
Works for me! Thanks

**Comment by Mawi137** [comment]
Released in 3.1.6!
